sparta/sparta/resources/Array.hpp

```diff
@@ -874,6 +874,15 @@ namespace sparta
             sparta_assert(idx < num_entries_,
                         "Cannot write to an index outside the bounds of the array.");
 
+            // We're overwriting an entry, clear valid and age information.
+            if(SPARTA_EXPECT_FALSE(isValid(idx)))
+            {
+                --num_valid_;
+                if(ArrayT == ArrayType::AGED) {
```

Comment:
Probably could get away with an 
'''
if constexpr 
'''
here, but that will be inconsistent with the rest of the code. (Unless you change them all 😜)